src/awkward/_broadcasting.py

```diff
@@ -1081,4 +1078,4 @@ def broadcast_and_apply(
         },
     )
     assert isinstance(out, tuple)
-    return tuple(broadcast_unpack(x, isscalar, backend) for x in out)
+    return tuple(broadcast_unpack(x, return_scalar, backend) for x in out)
```

Comment:
This is the main change; the caller determines whether the result should be a scalar.